### PR TITLE
[test] Unit test amount.h/amount.cpp

### PR DESCRIPTION
**Pull request by jimmysong**
Add tests for MoneyRange, binary operators, ToString and a constructor

### CODE REVIEW

File: src/test/amount_tests.cpp [hunk 1]
```diff
@@ -68,4 +80,28 @@ BOOST_AUTO_TEST_CASE(GetFeeTest)
...
+    a = CFeeRate(1);
+    b = CFeeRate(2);
+    BOOST_CHECK(a < b);
+    BOOST_CHECK(b > a);
+    BOOST_CHECK(a == a);
+    BOOST_CHECK(a <= b);
+    BOOST_CHECK(a <= a);
+    BOOST_CHECK(b >= a);
+    BOOST_CHECK(b >= b);
+    // a should now have 2 satoshis/kb
```
Comment (paveljanik) [inline src/test/amount_tests.cpp]: super-micro-nit: let's use our `ToString()` form here:
```
// a should be 0.00000002 BTC/kB now
```

**Comment by paveljanik** [comment]
Brilliant! Please fix my super-micro-nit and ACK

**Comment by jonasschnelli** [comment]
utACK d8a405bd94f0f609602bf55c80cf09f4d57d7d3c

**Comment by laanwj** [comment]
utACK

**Comment by jimmysong** [comment]
nit addressed and squashed.

**Comment by paveljanik** [comment]
ACK https://github.com/bitcoin/bitcoin/pull/10280/commits/dcb69fde5aae3ef572e299e57a45712f005daf8b

247 -> 250 tests. Thank you!

**Pull request merged**